### PR TITLE
Release FV3 solver updates

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**

Releasing some updates on behalf of Lucas Harris including:
Revision to semi-implicit solver to linearize vertical sound wave propagation about the hydrostatic state. This has been extensively tested by GFDL and EMC in SHiELD, GFS, and RRFS.

Fixes # (issue)

**How Has This Been Tested?**

Changes have been tested with Intel compiler using tests in SHiELD_build/RTS
This changes answers - The CI on this will not pass initially, then I will update reference answers and it will pass.

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by lharris4 — APPROVED** [review]
Looks good to me. Thanks for getting this approved so quickly.

Lucas

**Review by bensonr — COMMENTED** [review]
Looks like there are a few spacing changes from the old to the new code.  Can you look through them and reset as needed.

**Comment by KevinViner-NOAA** [comment]
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/issues/229

Great minds...

File: model/nh_utils.F90 [hunk 1]
```diff
@@ -338,8 +345,21 @@ subroutine Riem_Solver_c(ms,   dt,  is,   ie,   js, je, km,   ng,  &
...
+      RFw_initialized = .true.
+      do k=1,km
+         if (pfull(k) > rf_cutoff) exit
+         k_rf = k
+         rff(k) = dt/tau_w * sin(0.5*pi*log(rf_cutoff/pfull(k))/log(rf_cutoff/ptop))**2
+         rff(k) = 1.0d0 / ( 1.0d0+rff(k) )
+      enddo
+   endif
+
+
```
[the remark this answers is not in the document]
Comment (laurenchilutti) [reply model/nh_utils.F90:360]: Do you have a recommendation of either all double precision or all single?

**Comment by lharris4** [comment]
Hi, Rusty, Lauren. The forced double precision is from the Rayleigh damping
in fv_dynamics.F90. Since lower down (just above rf_cutoff) damping
timescales can get very long this is to avoid the damping coefficient from
getting rounded to 0. We can compute rff as double-precision too in the
previous line for consistency: this calculation is only done once and then
saved so efficiency isn't a concern.

Thanks,
Lucas

**Comment by lharris4** [comment]
Hi, Kate. With the separated variables tau_w and fast_tau_w_sec there
shouldn't be any conflicts, and the Rayleigh damping is completely separate
from the sound-wave change in SIM_solver().

Thanks,
Lucas

**Comment by laurenchilutti** [comment]
@bensonr your comments have been addressed - if you approve I will cherry-pick these final changes to the dev/emc PR

**Review by lharris4 — APPROVED** [review]
Looks fine to me. Thanks.

**Review by lharris4 — APPROVED** [review]
Thanks for making this name change too.

**Review by bensonr — APPROVED** [review]

**Pull request merged**